hw/ip/adc_ctrl/doc/_index.md

```diff
@@ -1,34 +1,215 @@
+In low power mode:
+* The ADC is only powered up for {{< regref "adc_pd_ctl.pwrup_time" >}} slow clock cycles plus the time taken for the ADC to take two samples plus NNN slow clock cycles.
+* The total time between samples for a periodic scan is the time the ADC is powered up plus {{< regref "adc_pd_ctl.wakeup_time" >}} cycles of the slow clock.
+* The time taken to transition to the continuous scanning mode is the total time of a slow scan multiplied by the {{< regref "adc_lp_sample_ctl" >}}.
+* The debounce timer will trigger the filter status (and interrupt) after the time taken to transition to fast mode plus {{< regref "adc_sample_ctl" >}} multiplied by the fast sampling cycle time.
+
+Although it can be used at any time, the periodic operation mode and use of the slow clock allows the ADC controller to continue to scan when most of the chip is in sleep or power-down modes.
+The controller can be configured to issue a wakeup to the rest of the chip.
+In this case if any filter enabled in the {{< regref "adc_wakeup_ctl" >}} register is hit then the wakeup will be issued.
+The filter that hit is recorded in the  {{< regref "adc_wakeup_status" >}} register and the bit should be cleared to acknowledge the wake has happened (no further wake events will be issued until this is done).
+
+
+## Filters and debounce
+
+There are two reserved bits in ADC filter control registers for future use. In the current implementation, ADC has 10-bit granularity. Each step is 2.148mV. It covers 0-2.2V.
+
+The ADC controller implements eight pairs of filters that feed the debounce logic.
+Each pair has a filter for channel 0 and a filter for channel 1.
+
+A filter consists of a `max` value, a `min` value and a `cond` flag indicating if the filter is hit by a sample inside or outside the range.
+* *Inside the range*: the filter is hit if `min` &le; `value` &le; `max`.
+* *Outside the range*: inverse of inside, so the filter is hit if `value` &lt; `min` or `value` &gt; `max`.
+
+Some example filters:
+* Inside `min=7`, `max=23`: any value between and including 7 and 23 will hit.
+* Outside `min=7`, `max=23`: any value less than 7 or greater than 23 will hit.
+* Inside `min=7`, `max=7`: the value must be exactly 7 to hit (sample noise may make an exact hit unlikely).
+* Inside `min=0`, `max=7`: the value must be less than 8 to hit.
+* Outside `min=8`, `max=0xFFF`: the value must be less than 8 to hit (alternate method).
+* Inside `min=0`, `max=0xFFF`: any value will hit. This may be useful to exclude one channel from the filter.
+* Outside `min=0`, `max=0xFFF`: no value will hit. If set for either channel the filter is effectively disabled.
+
+All pairs of filters that are enabled in {{< regref "adc_chn0_filter_ctl[7:0]" >}} and {{< regref "adc_chn1_filter_ctl[7:0]" >}} are evaluated after each pair of samples has been taken.
+The result is passed to the appropriate counter: the periodic scan counter if enabled and not at its limit otherwise the debounce counter.
```

Comment:
Nit:
I would insert some commas here or maybe slightly rephrase this sentence as:
```
The filter result is passed to the periodic scan counter, if it is enabled and not at its limit. 
Otherwise the result is passed to the debounce counter.

The list below describes how the counters interpret the filter results:
```

Reply:
Sure.